x-pack/plugins/cases/server/connectors/cases/cases_connector.ts

```diff
@@ -418,6 +458,115 @@ export class CasesConnector extends SubActionConnector<
+  }
+
+  private async reopenClosedCases(
+    casesClient: CasesClient,
+    closedCasesEntries: GroupedAlertsWithCases[],
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ): Promise<Map<string, GroupedAlertsWithCases>> {
+    const casesMapWithClosedCasesOpened = new Map(casesMap);
+
+    const bulkUpdateReq = closedCasesEntries.map((entry) => ({
+      id: entry.theCase.id,
+      version: entry.theCase.version,
+      status: CaseStatuses.open,
+    }));
+
+    /**
+     * TODO: bulkUpdate throws an error. Retry on errors.
+     */
+    const bulkUpdateCasesResponse = await casesClient.cases.bulkUpdate({ cases: bulkUpdateReq });
+
+    for (const res of bulkUpdateCasesResponse) {
+      if (casesMap.has(res.id)) {
+        const data = casesMap.get(res.id) as GroupedAlertsWithCases;
+        casesMapWithClosedCasesOpened.set(res.id, { ...data, theCase: res });
+      }
+    }
+
+    return casesMapWithClosedCasesOpened;
+  }
+
+  private async createNewCasesOutOfClosedCases(
+    params: CasesConnectorRunParams,
+    casesClient: CasesClient,
+    closedCasesEntries: GroupedAlertsWithCases[],
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ): Promise<Map<string, GroupedAlertsWithCases>> {
+    const casesMapWithClosedCasesOpened = new Map(casesMap);
+    const casesMapAsArray = Array.from(casesMap.values());
+
+    const findEntryByOracleRecord = (oracleId: string) => {
+      return casesMapAsArray.find((record) => record.oracleRecord.id === oracleId);
+    };
+
+    const bulkUpdateOracleValidRecords = await this.increaseOracleRecordCounter(
+      closedCasesEntries.map((entry) => entry.oracleRecord)
+    );
+
+    const groupedAlertsWithOracleRecords = new Map<string, GroupedAlertsWithOracleRecords>();
+
+    for (const record of bulkUpdateOracleValidRecords) {
+      const foundRecord = findEntryByOracleRecord(record.id);
+
+      if (foundRecord) {
+        groupedAlertsWithOracleRecords.set(record.id, {
+          oracleKey: record.id,
+          oracleRecord: foundRecord.oracleRecord,
+          alerts: foundRecord.alerts,
+          grouping: foundRecord.grouping,
+        });
+      }
+    }
+
+    const groupedAlertsWithCaseId = this.generateCaseIds(params, groupedAlertsWithOracleRecords);
+    const bulkCreateReq = Array.from(groupedAlertsWithCaseId.values()).map((record) =>
+      this.getCreateCaseRequest(params, record)
+    );
+
+    /**
+     * TODO: bulkCreate throws an error. Retry on errors.
+     */
+    const bulkCreateCasesResponse = await casesClient.cases.bulkCreate({ cases: bulkCreateReq });
+
+    for (const res of bulkCreateCasesResponse.cases) {
+      if (groupedAlertsWithCaseId.has(res.id)) {
+        const data = groupedAlertsWithCaseId.get(res.id) as GroupedAlertsWithCaseId;
+        casesMapWithClosedCasesOpened.set(res.id, { ...data, theCase: res });
+      }
+    }
+
+    return casesMapWithClosedCasesOpened;
```

Comment:
nit: but since this method is `createNewCasesOutOfClosedCases` this var should probably be named something else. At least initially I wasn't sure where I was when I read the variable name 😅

Reply:
Ok! Any suggestions? The map contains all cases so far (closed and open) plus the new ones that were created.

Reply:
Hmm if the other method is `reopenClosedCases` and the result is `casesMapWithClosedCasesOpened` and this is `createNewCasesOutOfClosedCases`, we could call it `casesMapWithNewCases` something like that.

I don't mind either way, just not the same name :D